pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go

```diff
@@ -345,6 +347,11 @@ func newProvisioner(plugin volume.ProvisionableVolumePlugin, claim *api.Persiste
 		AccessModes:                   claim.Spec.AccessModes,
 		PersistentVolumeReclaimPolicy: api.PersistentVolumeReclaimDelete,
 		CloudTags:                     &tags,
+		ClusterName:                   controller.clusterName,
+	}
+
+	if pv != nil {
+		volumeOptions.PVName = pv.Name
```

Comment:
What kind of name is guaranteed to be unique in the cluster besides a new UUID?  objMeta.GenerateName allows a human-readable prefix with a suffix that makes a unique name.

Also, once the PV is persisted, we get more than a name.  We get a marker (with a future Provisioning Condition) that makes the provisioning process async.  We use annotations today but that is better expressed as a Condition (can put meaningful status on the CLI).